test/jdk/jdk/jfr/tool/TestMetadata.java

```diff
@@ -75,4 +87,65 @@ public static void main(String[] args) throws Throwable {
             lineNumber++;
         }
     }
-}
+
+    static void testEventTypeNum() throws Throwable {
+        OutputAnalyzer output = ExecuteHelper.jfr("metadata");
+        List<String> eventNames = new ArrayList<>();
+        List<String> lines = output.asLines();
+
+        for (String line : lines) {
+            if (line.startsWith("@Name(\"")) {
+                eventNames.add(line.substring(7, line.indexOf("\"", 7)));
+            }
+        }
+        List<EventType> eventTypes = FlightRecorder.getFlightRecorder().getEventTypes();
+        List<String> expectedNames = new ArrayList<>();
+        for (EventType eventType : eventTypes) {
+            expectedNames.add(eventType.getName());
+        }
+        Asserts.assertGTE(eventNames.size(), expectedNames.size());
```

Comment:
Seems something like this would be sufficient:

    public void testNumberOfEventTypes() {
      int count = 0;
      for (String line : output.asLines()) {
      if (line.contains("extends jdk.jfr.Event")) {
        count++;
      }
      Assert.assertEquals(count, FlightRecorder.getFlightRecorder().getEventTypes().size());
    }
≈